scr/src/main/java/org/apache/felix/scr/impl/ComponentRegistry.java

```diff
@@ -774,4 +774,11 @@ public void run()
             }
         }
     }
+
+    public void shutdown() {
```

Comment:
Should this not be guarded with `synchronized ( this.changeCountTimerLock ) {` and probably set the field to null?

Reply:
I don't think so.  `changeCountTimer` is volatile here and the object itself is thread safe.  Calling cancel multiple times should not be an issue.  Besides that once ComponentRegistry is shutdown references to it are gone from the activator and it should GC just fine.  Technically even the activator doesn't need to null out the registry there because the framework throws away the activator instance after the bundle is stopped.

Also, nulling out would incorrectly allow a new timer if the method `org.apache.felix.scr.impl.ComponentRegistry.updateChangeCount()` is somehow called again from another thread after the bundle has stopped.